drizzlepac/haputils/product.py

```diff
@@ -127,6 +129,220 @@ def generate_metawcs(self):
+            mosaic_catalog_list = alignment_pars['run_align']['mosaic_catalog_list']
+        else:
+            mosaic_catalog_list = catalog_list
+        num_cat = len(mosaic_catalog_list)
+
+        # Fitting methods
+        mosaic_method_list = alignment_pars['run_align']['mosaic_fit_list']
+        num_method = len(mosaic_method_list)
+
+        # Fit geometry methods
+        # mosaic_fitgeom_list = list(alignment_pars['run_align']['mosaic_fitgeom_list'][0].items())
+        mosaic_fitgeom_list = list(alignment_pars['run_align']['mosaic_fitgeom_list'].items())
+
+        for edp in self.edp_list:
+            exposure_filenames.append(edp.full_filename)
+            headerlet_filenames[edp.full_filename] = edp.headerlet_filename
+            crclean.append(edp.crclean)
+
+        try:
+            # Proceed only if there is data to process
+            if self.edp_list:
+
+                # If necessary, generate the alignment table only once
+                if align_table is None:
+                    align_table = align_utils.AlignmentTable(exposure_filenames,
+                                                             log_level=self.log_level,
+                                                             **alignment_pars)
+                    align_table.find_alignment_sources(output=output, crclean=crclean)
+
+                is_good_fit = False
+
+                # Loop for available catlogs, the first successful fit for a
+                # (catalog, fitting method, and fit geometry) is satisfactory to break out of the looping.
+                for index_cat, catalog_item in enumerate(mosaic_catalog_list):
+
+                    more_catalogs = True
+                    if (index_cat + 1) == num_cat:
+                        more_catalogs = False 
+
+                    # Override the default self.refname as it really needs to be
+                    # catalog-specific to be useful
+                    self.refname = self.product_basename + "_" + catalog_item + "_ref_cat.ecsv"
+
+                    log.info("Starting alignment to absolute astrometric reference frame '{}'.".format(catalog_item))
+                    log.debug('Creating reference catalog {}'.format(self.refname))
+                    ref_catalog = amutils.create_astrometric_catalog(align_table.process_list,
+                                                                     catalog=catalog_item,
+                                                                     output=self.refname,
+                                                                     gaia_only=False,
+                                                                     full_catalog=True)
+
+                    # Add a weight column which is based upon proper motion measurements
+                    if ref_catalog.meta['converted'] and ref_catalog['RA_error'][0] != np.nan:
+                        ref_weight = 1. / np.sqrt(ref_catalog['RA_error'] ** 2 + ref_catalog['DEC_error'] ** 2)
```

Comment:
The definition of the weight should be inverted for best results.  Please remove '1. /' from this line.

Reply:
Issue corrected.